processing/src/main/java/io/druid/segment/IndexIO.java

```diff
@@ -251,6 +247,99 @@ public static boolean convertSegment(File toConvert, File converted, IndexSpec i
     public MMappedIndex mapDir(File inDir) throws IOException;
   }
 
+  public static void validateRowValues(
+      Rowboat rb1,
+      IndexableAdapter adapter1,
```

Comment:
For anyone else reading this: the adapter is used for two things:

1. To get the dimension names
2. To get the indexed strings.

In order to not use the adapter but use the indexed<string> directly, I would still need a map from dim position to dim index, which is effectively what the adapter does.